src/NUnitFramework/framework/Internal/ThreadUtility.cs

```diff
@@ -33,22 +36,82 @@ namespace NUnit.Framework.Internal
     /// </summary>
     public static class ThreadUtility
     {
+        private const int ThreadAbortedCheckDelay = 100;
+
+
+        /// <summary>
+        /// Pre-Task compatibility
+        /// </summary>
+        public static void Delay(int milliseconds, WaitCallback threadPoolWork, object state = null)
+        {
+            new DelayClosure(milliseconds, threadPoolWork, state);
+        }
+
+        private sealed class DelayClosure
+        {
+            private readonly Timer _timer;
+            private readonly WaitCallback _threadPoolWork;
+            private readonly object _state;
+
+            public DelayClosure(int milliseconds, WaitCallback threadPoolWork, object state)
+            {
+                _threadPoolWork = threadPoolWork;
+                _state = state;
+                _timer = new Timer(Callback, null, Timeout.Infinite, Timeout.Infinite);
+                // Make sure _timer is set before starting the timer so that it is disposed.
```

Comment:
I should leave a note in the source mentioning the possible race condition if the field is not set before allowing the timer to call back.